src/Raven.Server/Documents/Indexes/Static/IndexedDocumentsEnumerator.cs

```diff
@@ -15,22 +14,53 @@ public IndexedDocumentsEnumerator(IEnumerable<Document> docs)
 
         public IEnumerator<DynamicDocumentObject> GetEnumerator()
         {
-            Document previous = null;
+            return new DynamicObjectEnumerator(_docs);
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return new DynamicObjectEnumerator(_docs);
+        }
+
+        private class DynamicObjectEnumerator : IEnumerator<DynamicDocumentObject>
+        {
+            private readonly DynamicDocumentObject _dynamicDocument = new DynamicDocumentObject();
+            private readonly IEnumerator<Document> _inner;
+            private Document _previous;
+
+            public DynamicObjectEnumerator(IEnumerable<Document> docs)
+            {
+                _inner = docs.GetEnumerator();
+            }
 
-            foreach (var doc in _docs)
+            public bool MoveNext()
             {
-                previous?.Data.Dispose();
+                if (_inner.MoveNext() == false)
+                    return false;
```

Comment:
Good point. It should be disposed in Dispose of the enumerator. I will fix that in a moment.